### PR TITLE
[18068] Monitor Service test plan implementation

### PR DESCRIPTION
**Pull request by Mario-DL**
## Description

This PR adds the test plan implementation for the `Monitor Service` .

## Contributor Checklist
- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- **N/A**  Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [X] Any new/modified methods have been properly documented using Doxygen. 
- [X] Changes are ABI compatible. 
- [X] Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 
    
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by Mario-DL** [comment]
@richiprosima please test this

File: test/blackbox/common/DDSBlackboxTestsMonitorService.cpp [hunk 1]
```diff
@@ -0,0 +1,1761 @@
...
+#ifdef FASTDDS_STATISTICS
+    //! Setup
+    MonitorServiceParticipant MSP;
+
+    //! Procedure
+    MSP.setup();
+
+    //! Assertions
+    ASSERT_EQ(ReturnCode_t::RETCODE_NOT_ENABLED, MSP.disable_monitor_service());
+    ASSERT_EQ(ReturnCode_t::RETCODE_OK, MSP.enable_monitor_service());
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply test/blackbox/common/DDSBlackboxTestsMonitorService.cpp:874]: Yes ! it is possible. In fact, it is tested in the `DDS-MS-SIMPLE-11`.

File: test/blackbox/common/DDSBlackboxTestsMonitorService.cpp [hunk 2]
```diff
@@ -0,0 +1,1761 @@
...
+        "monitor_service_property_participant", "monitor_service_statistics_property_participant" };
+
+    MonitorServiceParticipant MSP;
+
+    //! Procedure
+    MSP.setup(xml_file, participant_profile_names.first);
+    ASSERT_EQ(ReturnCode_t::RETCODE_OK, MSP.disable_monitor_service());
+
+    MSP.reset();
+    MSP.setup(xml_file, participant_profile_names.second);
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply test/blackbox/common/DDSBlackboxTestsMonitorService.cpp:900]: Yes, adding it to the test plan.

File: test/blackbox/common/DDSBlackboxTestsMonitorService.cpp [hunk 3]
```diff
@@ -0,0 +1,1761 @@
...
+                cv.notify_one();
+            }
+        }
+    }
+
+};
+
+struct ConnectionListSampleValidator : public SampleValidator
+{
+    void validate(
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply test/blackbox/common/DDSBlackboxTestsMonitorService.cpp:560]: I think I am better defining the exact locators to be used in a custom transport descriptor.

File: test/blackbox/common/DDSBlackboxTestsMonitorService.cpp [hunk 4]
```diff
@@ -0,0 +1,1761 @@
...
+ * Multiple MSC shall correctly receive the updates after late joining.
+ */
+TEST(DDSMonitorServiceTest, monitor_service_advanced_late_joiners)
+{
+#ifdef FASTDDS_STATISTICS
+    //! Setup
+    size_t n_participants = 3;
+    std::vector<MonitorServiceParticipant> MSPs;
+    MSPs.reserve(n_participants);
+    MonitorServiceConsumer MSC;
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply test/blackbox/common/DDSBlackboxTestsMonitorService.cpp:1675]: Yes, the idea here is to have 3 MSPs and 1 MSC. Nonetheless, I need to include a fix in the test-plan to make a single one from the DDS-MS-ADV-03 and DDS-MS-ADV-04 specified there

File: test/blackbox/common/DDSBlackboxTestsMonitorService.cpp [hunk 5]
```diff
@@ -591,6 +676,72 @@ struct ConnectionListSampleValidator : public SampleValidator
...
+
+                                                return same_locators;
+                                            }
+                                            else
+                                            {
+                                                return false;
+                                            }
+                                        });
+
+                                        if (conn_it != elem.value().connection_list().end())
```
Comment (jsantiago-eProsima) [inline test/blackbox/common/DDSBlackboxTestsMonitorService.cpp:726]: Maybe I'm not following the nested lambdas correctly but, if the find_if succeeds then doesn't that mean that the locators are the expected ones?

File: test/blackbox/common/DDSBlackboxTestsMonitorService.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (jsantiago-eProsima) [reply test/blackbox/common/DDSBlackboxTestsMonitorService.cpp:1892]: Then the description "Multiple MSC shall correctly receive..." does not properly portray what is tested.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: test/blackbox/common/DDSBlackboxTestsMonitorService.cpp (hunk 5 above)
Comment (Mario-DL) [reply test/blackbox/common/DDSBlackboxTestsMonitorService.cpp:726]: Good catch ! the condition was just the opposite

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

**Review by jsantiago-eProsima — APPROVED** [review]

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Comment by Mario-DL** [comment]
@richiprosima please test this

**Comment by Mario-DL** [comment]
@richiprosima please test mac

**Comment by Mario-DL** [comment]
@richiprosima please test mac

**Pull request merged**